File: src/Implicit/ImplicitBias.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitBias : ImplicitModuleBase
    {
        public ImplicitBias(ImplicitModuleBase source, double bias)
        {
            this.Source = source;
            this.Bias = new ImplicitConstant(bias);
        }

        public ImplicitBias(ImplicitModuleBase source, ImplicitModuleBase bias)
        {
            this.Source = source;
            this.Bias = bias;
        }

        public ImplicitModuleBase Source { get; set; }

        public ImplicitModuleBase Bias { get; set; }

        public override double Get(double x, double y)
        {
            return Utilities.Bias(this.Bias.Get(x, y), this.Source.Get(x, y));
        }

        public override double Get(double x, double y, double z)
        {
            return Utilities.Bias(this.Bias.Get(x, y, z), this.Source.Get(x, y, z));
        }

        public override double Get(double x, double y, double z, double w)
        {
            return Utilities.Bias(this.Bias.Get(x, y, z, w), this.Source.Get(x, y, z, w));
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return Utilities.Bias(this.Bias.Get(x, y, z, w, u, v), this.Source.Get(x, y, z, w, u, v));
        }
    }
}

File: src/Implicit/ImplicitBias.java
package Implicit;

import util.Maths;

public final class ImplicitBias extends ImplicitModuleBase {
    public ImplicitBias(final ImplicitModuleBase source, final double bias) {
        this.Source = source;
        this.Bias = new ImplicitConstant(bias);
    }

    public ImplicitBias(final ImplicitModuleBase source, final ImplicitModuleBase bias) {
        this.Source = source;
        this.Bias = bias;
    }

    private ImplicitModuleBase Source;

    private ImplicitModuleBase Bias;

    public ImplicitModuleBase getSource() {
        return this.Source;
    }

    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }

    public ImplicitModuleBase getBias() {
        return this.Bias;
    }

    public void setBias(final ImplicitModuleBase bias) {
        this.Bias = bias;
    }

    @Override
    public double Get(final double x, final double y) {
        return Maths.Bias(this.Bias.Get(x, y), this.Source.Get(x, y));
    }

    @Override
    public double Get(final double x, final double y, final double z) {
        return Maths.Bias(this.Bias.Get(x, y, z), this.Source.Get(x, y, z));
    }

    @Override
    public double Get(final double x, final double y, final double z, final double w) {
        return Maths.Bias(this.Bias.Get(x, y, z, w), this.Source.Get(x, y, z, w));
    }

    @Override
    public double Get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return Maths.Bias(this.Bias.Get(x, y, z, w, u, v), this.Source.Get(x, y, z, w, u, v));
    }
}
